MISRA.md

```diff
@@ -0,0 +1,26 @@
+# MISRA Compliance
+
+The MQTT library files conform to the [MISRA C:2012](https://www.misra.org.uk/MISRAHome/MISRAC2012/tabid/196/Default.aspx)
+guidelines, with some noted exceptions. Compliance is checked with Coverity static analysis.
+Deviations from the MISRA standard are listed below:
+
+### Ignored by Coverity Configuration
```

Comment:
Where is the configuration stored?  Is it in the repository?

Reply:
Good question! I have created a PR #44 to add the config in this repository:

Reply:
No, the configuration is in the hub repository: https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/tools/coverity/misra.config. Should it be linked here?

Reply:
Added a link to the file